### PR TITLE
[bitnami/keycloak] add support to disable service links

### PR DESCRIPTION
**Pull request by nkreiger**
### Description of the change

Adds configuration option to disable kubernetes service links. 

This is useful for naming conflicts and to resolve `too many arg` issues if you are overloading with too much in one namespace. Which you can be forced to do sometimes...First contrib, hopefully I followed standards correctly, sorry if I didn't.

### Checklist

- [x ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/keycloak/values.yaml [hunk 1]
```diff
@@ -35,6 +35,12 @@ namespaceOverride: ""
 ## @param commonLabels Labels to add to all deployed objects
 ##
 commonLabels: {}
+## @param serviceLinks Boolean to disable kubernetes service links
+##
+serviceLinks:
+  ## @param serviceLinks.disabled Disable kubernetes service links (enabled by default in kubernetes)
+  ##
+  disabled: false
```
Comment (carrodher) [inline bitnami/keycloak/values.yaml:43]: Thanks for your contribution, there is an integration that is failing due to an issue in the metadata used to generate the README. Could you apply something like this?
```suggestion
## @param serviceLinks.disabled Disable kubernetes service links (enabled by default in kubernetes)
##
serviceLinks:
  disabled: false
```

File: bitnami/keycloak/templates/statefulset.yaml [hunk 2]
```diff
@@ -72,6 +72,9 @@ spec:
       {{- if .Values.podSecurityContext.enabled }}
       securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
       {{- end }}
+      {{- if .Values.serviceLinks.disabled }}
+      enableServiceLinks: false
+      {{- end }}
```
Comment (migruiz4) [inline bitnami/keycloak/templates/statefulset.yaml:77]: `enableServiceLinks` is only available in Kubernetes 1.13 and later, to keep backward compatibility, I suggest the following change:
```suggestion
      {{- if semverCompare ">= 1.13" (include "common.capabilities.kubeVersion" .) }}
      enableServiceLinks: {{ .Values.enableServiceLinks }}
      {{- end }}
```

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @nkreiger,

Thank you very much for your contribution. Could you please take a look at my suggestion?

I think the value `serviceLinks.disabled` to configure `enableServiceLinks` could be a bit confusing. 

My suggestion is to use the same name as the Kubernetes attribute, to make it intuitive and easier for users to find.

File: bitnami/keycloak/Chart.yaml [hunk 3]
```diff
@@ -26,4 +26,4 @@ name: keycloak
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/keycloak
   - https://github.com/keycloak/keycloak
-version: 13.1.3
+version: 13.1.4
```
Comment (migruiz4) [inline bitnami/keycloak/Chart.yaml:29]: Hi @nkreiger,

As this is PR introduces a new feature, could you please bump a minor version? (13.2.0)

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thank you very much for applying the suggested changes, looks good to me!

Just one last change before we can merge this PR. Could you please bump a minor version?

File: bitnami/keycloak/Chart.yaml (hunk 3 above)
Comment (nkreiger) [reply bitnami/keycloak/Chart.yaml:29]: sure!

**Review by migruiz4 — APPROVED** [review]
Looks good to me, thank you for your contribution!

**Pull request merged**